helix-ui/src/main/java/org/apache/helix/ui/api/ResourceSpec.java

```diff
@@ -0,0 +1,31 @@
+package org.apache.helix.ui.api;
+
+public class ResourceSpec extends ClusterSpec {
+    private final String resourceName;
+
+    public ResourceSpec(String zkAddress, String clusterName, String resourceName) {
+        super(zkAddress, clusterName);
+        this.resourceName = resourceName;
+    }
+
+    public String getResourceName() {
+        return resourceName;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (!(o instanceof ResourceSpec)) {
+            return false;
+        }
+        ResourceSpec c = (ResourceSpec) o;
+        return getZkAddress().equals(c.getZkAddress())
+                && getClusterName().equals(c.getClusterName())
+                && resourceName.equals(c.getResourceName());
+    }
+
+    @Override
+    public int hashCode() {
+        return super.hashCode() + 27 * resourceName.hashCode();
```

Comment:
Probably a good idea, but I didn't know about it while writing this. Do you think this should be changed? I don't know if it really matters in this case because we're not putting a million of them in a HashMap or anything.